src/deploy/osp_deployer/director.py

```diff
@@ -553,7 +558,15 @@ def setup_environment(self):
                     logger.warning(
                         "Bad entry in osd_disks: {}".format(osd))
 
+        if osds_per_node < 1:
+            logger.info("no OSDs available. exiting")
+            sys.exit(1)
+
         total_osds = self.calc_num_osds(osds_per_node)
+        if total_osds == 0:
+            logger.info("no OSDs available. exiting")
```

Comment:
Could you beef up the logged message a little?  Maybe say something like: Either the OSD configuration is not specified in the .properties file, or the storage nodes have no available storage to dedicate to OSDs.  Or something along those lines so the user has an idea of how to correct the problem.

Reply:
I expanded the message as suggested.